File: DonRyu/2022-11-03-Rotate Image.js
var rotate = function(matrix) {
    // Firstly Transpose The Matrix
   for (let i = 0 ; i < matrix.length;i++){
       for (let j = i ; j < matrix.length;j++){
           var Temp = matrix[j][i]
           matrix[j][i] = matrix[i][j]
           matrix[i][j] = Temp
       }
   }
   // Secondly Make Reflected Image Of Matrix
   for (let i = 0 ; i < matrix.length;i++){
       for (let j = 0 ; j < matrix.length / 2 ;j++){
           var Temp = matrix[i][j]
           matrix[i][j] = matrix[i][matrix.length-j-1]
           matrix[i][matrix.length - j - 1] = Temp
       }
   }

};

File: BrianLee/Week8/2022-11-03-Rotate-Image.java
class Solution {
    public void rotate(int[][] matrix) {
        int up = 0;
        int down = matrix.length-1;
        while(up < down) {
            for(int j = 0; j < matrix[0].length; j++) {
                int temp = matrix[up][j];
                matrix[up][j] = matrix[down][j];
                matrix[down][j] = temp;
            }
            up++;
            down--;
        }

        for(int i = 0; i < matrix.length; i++) {
            for(int j = 0; j < i; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
    }
}